libraries/botbuilder-applicationinsights/src/applicationInsightsTelemetryClient.ts

```diff
@@ -0,0 +1,100 @@
+import { BotTelemetryClient, Severity } from 'botbuilder-core';
+const appInsights = require('applicationinsights');
+const cls = require('cls-hooked');
+const ns = cls.createNamespace('my.request');
+
+// This is the currently recommended work-around for using Application Insights with async/await
+// https://github.com/Microsoft/ApplicationInsights-node.js/issues/296
+// This allows AppInsights to automatically apply the appropriate context objects deep inside the async/await chain.
+const correlationContextManager = require('applicationinsights/out/AutoCollection/CorrelationContextManager').CorrelationContextManager;
+const origGetCurrentContext = correlationContextManager.getCurrentContext;
+
+function getCurrentContext(){
+  return ns.get('ctx') || origGetCurrentContext();
+}
+
+// Overwrite the built-in getCurrentContext() method with a new one.
+correlationContextManager.getCurrentContext = getCurrentContext;
+
+export const ApplicationInsightsWebserverMiddleware = function (req, res, next) {
+
+  // Check to see if the request contains an incoming request.
+  // If so, set it into the Application Insights context.
+  const activity = req.body;
+  if (activity && activity.id) {
+    const context = appInsights.getCorrelationContext();
+    context.activity = req.body;
+  }
+
+  ns.bindEmitter(req);
+  ns.bindEmitter(res);
+  ns.run(function(){
+    ns.set('ctx', origGetCurrentContext());
+    next();
+  });
+
+};
+
+
+export class ApplicationInsightsTelemetryClient implements BotTelemetryClient {
+
+    private _telemetryClient: any;
+
+    // TODO: Do we want to expose the post-setup options for developers to configure themselve?
+    constructor(settings: any) {
+
+        appInsights.setup(settings)
+        .setAutoDependencyCorrelation(true)
+        .setAutoCollectRequests(true)
+        .setAutoCollectPerformance(true)
+        .setAutoCollectExceptions(true)
+        .setAutoCollectDependencies(true)
+        .setAutoCollectConsole(true, true)

```

Comment:
>setAutoCollectConsole [](start = 9, length = 21)

For C# side, it's common to integrate with other systems for tracing stuff.  By default on the C# side, we're not consuming tracing - it will be in the sample to show how to do this.  Not sure if there's a similar thing going on in nodejs..

Reply:
As is, this captures the built in console.log mechanism AND the common custom loggers.

I will have to see if this option can be changed after the telemetry collection system is initialized...

Reply:
This is now possible - the configuration object, upon which these config methods can be called - is exposed at appInsightsClient.configuration, so further tuning of the options can take place.